### PR TITLE
chore: small `ChaindataProvider` refactor

### PR DESCRIPTION
**Pull request by alecdwm**

### CODE REVIEW

File: packages/chaindata-provider/src/ChaindataProvider.ts [hunk 1]
```diff
@@ -1,37 +1,700 @@
...
+  }
+
+  //
+  // items by genesisHash
+  //
+
+  get chainsByGenesisHashObservable() {
+    return this.chainsObservable.pipe(map(util.itemsToMapByGenesisHash))
+  }
+  get chainsByGenesisHash() {
```
Comment (0xKheops) [inline packages/chaindata-provider/src/ChaindataProvider.ts]: not super comfortable with async properties, callers may think they can call it multiple times without impact. 
would prefer a getChainsByGenesisHash() method.
wdyt ?

Comment (alecdwm) [reply packages/chaindata-provider/src/ChaindataProvider.ts]: I think that's a good idea, I'll change them all to be getXXX methods

Comment (alecdwm) [reply packages/chaindata-provider/src/ChaindataProvider.ts]: Done

**Review by 0xKheops — DISMISSED** [review]
great small refactor :)

**Review by chidg — APPROVED** [review]
LGTM

**Pull request merged**